pkcs11/src/config/device.rs

```diff
@@ -290,8 +310,8 @@ impl InstanceData {
             }
         };
         drop(write);
-        if let Some(t) = &*RETRY_THREAD.load() {
-            t.send(RetryThreadMessage::FailedInstnace {
+        if let Some(t) = &*RETRY_THREAD.read().unwrap() {
```

Comment:
Can we really unwrap these errors?  At first glance, it looks like there could be a race condition when `stop_background_timer` is called.

Reply:
And the lock can only be poisoned when a `Write` handle is heald, and the only times this is the case the operations are panic-free (unless there is a bug in the lock itself).

Reply:
There could be a panic in the background threads though. It might make sense to just log the error when calling `.join` in their `JoinHandle`.